### PR TITLE
feat: Implement enqueue accepted wholesale services from Process Manager

### PR DESCRIPTION
**Pull request by ebbeknudsen**
## Description

- Implement enqueue accepted wholesale services from Process Manager
- Implement enqueue rejected wholesale services when no data is found
- Add function test for enqueue accepted wholesale services
- Add function test for enqueue rejected wholesale services when no data is found
- Remove unused `RequestWholesaleServices` durable function
- Update Process Manager package
- Copy `CalculatedQuantityQualityMapper` and tests from process module

**Behaviour tests will be updated later, which should test edge cases, mappings etc.**

## Checklist
- [ ] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [ ] Has it been considered whether data is being delivered to the wrong actor?
- [ ] Subsystem test executed (dev_002/dev_003)
- [ ] Is there time to monitor state of the release to Production?
- [ ] Reference to the task

## Copilot summary

This pull request includes several updates and improvements to the B2B API application tests and related functionality. The most important changes involve updating package references, adding new test data, and enhancing the message handling logic in the tests.

### Package updates:
* Updated `Energinet.DataHub.ProcessManager.Abstractions` to version `0.26.2` and `Energinet.DataHub.ProcessManager.Orchestrations.Abstractions` to version `0.18.4` in `B2BApi.AppTests.csproj` and `B2BApi.csproj` files. [[1]](diffhunk://#diff-efe8e992814a8d657d423ae0107b9d0514523092046ef1f63ea38d039556cffdL12-R13) [[2]](diffhunk://#diff-aa130eb2b0ff919e9be693d087a9a1853b511bdd57779e4346b46bae3a4269a2L28-R30)

### Test improvements:
* Added `ResetMessageHandlersAndReceivedMessages` calls to `InitializeAsync` and `DisposeAsync` methods in `EnqueueBrs026MessagesTests` and `EnqueueBrs028MessagesTests` to ensure clean test state. [[1]](diffhunk://#diff-0c2362d83cf15b6c8f3eefff68ee5e898d6bbda2a2d04667bd349f60139f5e82R57-R63) [[2]](diffhunk://#diff-4672a3ee137ffca203a7fc40403ee3f50dc654ee7aed6efa733675e4d90d2bf7R59-R192)
* Enhanced `EnqueueBrs028MessagesTests` with new test cases and improved assertions for verifying enqueued messages and ServiceBus notifications. [[1]](diffhunk://#diff-4672a3ee137ffca203a7fc40403ee3f50dc654ee7aed6efa733675e4d90d2bf7R59-R192) [[2]](diffhunk://#diff-4672a3ee137ffca203a7fc40403ee3f50dc654ee7aed6efa733675e4d90d2bf7L150-R214) [[3]](diffhunk://#diff-4672a3ee137ffca203a7fc40403ee3f50dc654ee7aed6efa733675e4d90d2bf7R253-R321)

### New test data:
* Introduced `AmountsPerChargeTestDataDescription` class to provide structured test data for calculation results, including a predefined result set for testing.

### Code enhancements:
* Added `IUnitOfWork` dependency to `EnqueueHandler_Brs_026_V1` and ensured transaction commit in `EnqueueRejectedMessagesAsync` method. [[1]](diffhunk://#diff-f185f02595a6de8aea97ef89ca35e56e051da2a84a0eac513e3ffbfee2804c02R16) [[2]](diffhunk://#diff-f185f02595a6de8aea97ef89ca35e56e051da2a84a0eac513e3ffbfee2804c02L37-R45) [[3]](diffhunk://#diff-f185f02595a6de8aea97ef89ca35e56e051da2a84a0eac513e3ffbfee2804c02R125-R126)

### CODE REVIEW

File: source/B2BApi.AppTests/TestData/CalculationResults/AmountsPerChargeTestDataDescription.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    public static readonly Instant PeriodEnd = Instant.FromUtc(2022, 01, 31, 23, 00);
+
+    public static readonly ResultSet ResultSet1 = new(
+        PeriodStart: PeriodStart,
+        PeriodEnd: PeriodEnd,
+        GridArea: "804",
+        BusinessReason: BusinessReason.WholesaleFixing,
+        EnergySupplierNumber: ActorNumber.Create("5790001687137"),
+        RowCount: 837,
+        ExpectedMessagesCount: 27); // Resolution is daily, so 283 rows / 31 days = 27 expected messages.
```
Comment (RasmusGraabaek) [inline source/B2BApi.AppTests/TestData/CalculationResults/AmountsPerChargeTestDataDescription.cs:32]:
```suggestion
        ExpectedMessagesCount: 27); // Resolution is daily, so 837 rows / 31 days = 27 expected messages.
```

**Review by lasrinnil — APPROVED** [review]

File: source/B2BApi/Functions/EnqueueMessages/BRS_028/EnqueueHandler_Brs_028_V1.cs [hunk 2]
```diff
@@ -74,29 +141,106 @@ protected override async Task EnqueueRejectedMessagesAsync(
...
+        if (businessReason == BusinessReason.BalanceFixing)
+            return CalculationType.BalanceFixing;
+        if (businessReason == BusinessReason.PreliminaryAggregation)
+            return CalculationType.Aggregation;
+        if (businessReason == BusinessReason.WholesaleFixing)
+            return CalculationType.WholesaleFixing;
+        if (businessReason == BusinessReason.Correction)
+        {
+            if (settlementVersion is null)
+                return null; // CalculationType == null means get latest correction
```
[the remark this answers is not in the document]
Comment (ebbeknudsen) [reply source/B2BApi/Functions/EnqueueMessages/BRS_028/EnqueueHandler_Brs_028_V1.cs:185]: Do you mean a `SettlementVersion.LatestCorrection` so you can do:
```csharp
if (settlementVersion == SettlementVersion.LatestCorrection)
	return null;
```

Or a `CalculationType.LatestCorrection` so you can do:
```csharp
if (settlementVersion == null)
	return CalculationType.LatestCorrection;
```

?

File: source/B2BApi/Functions/EnqueueMessages/BRS_028/EnqueueHandler_Brs_028_V1.cs [hunk 3]
```diff
@@ -47,7 +55,66 @@ protected override async Task EnqueueAcceptedMessagesAsync(
             "Received enqueue accepted message(s) for BRS 028. Data: {0}",
             acceptedData);
 
-        // TODO: Call actual logic that enqueues accepted messages instead
+        var queryParams = new WholesaleServicesQueryParameters(
+            AmountType: AmountType.AmountPerCharge, // Default to AmountPerCharge, will be replaced in foreach loop
```
[the remark this answers is not in the document]
Comment (ebbeknudsen) [reply source/B2BApi/Functions/EnqueueMessages/BRS_028/EnqueueHandler_Brs_028_V1.cs:59]: No, it should never be null where the query parameters is used. This is just a "hack" to avoid having the create/map the type twice in the 2 code paths, when only the `AmountType` is changed.

**Review by RasmusGraabaek — APPROVED** [review]
